src/Xamarin.Android.Build.Tasks/Tasks/GenerateCompressedAssembliesNativeSourceFiles.cs

```diff
@@ -61,7 +64,9 @@ void GenerateCompressedAssemblySources ()
 				kvp.Value.DescriptorIndex = index++;
 			}
 
-			BuildEngine4.RegisterTaskObject (CompressedAssemblyInfo.CompressedAssembliesInfoKey, assemblies, RegisteredTaskObjectLifetime.Build, false);
+			string key =CompressedAssemblyInfo.GetKey (ProjectFullPath);
```

Comment:
You might just fix when updating this to target monodroid/master:
```suggestion
			string key = CompressedAssemblyInfo.GetKey (ProjectFullPath);
```